rules/generic/src/Rector/MethodCall/MethodCallRemoverRector.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Generic\Rector\MethodCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\Variable;
+use Rector\Core\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\ConfiguredCodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+
+/**
+ * @see \Rector\Generic\Tests\Rector\MethodCall\MethodCallRemoverRector\MethodCallRemoverRectorTest
+ */
+final class MethodCallRemoverRector extends AbstractRector implements ConfigurableRectorInterface
+{
+    public const METHOD_CALL_REMOVER_ARGUMENT = '$methodCallRemoverArgument';
+
+    /**
+     * @var string[]
+     */
+    private $methodCallRemoverArgument;
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Turns "$this->something()->anything()" to "$this->anything()"', [
+            new ConfiguredCodeSample(
+                <<<'PHP'
+$someObject = new Car;
+$someObject->something()->anything();
+PHP
+                ,
+                <<<'PHP'
+$someObject = new Car;
+$someObject->anything();
+PHP
+                ,
+                [
+                    self::METHOD_CALL_REMOVER_ARGUMENT => [
+                        self::METHOD_CALL_REMOVER_ARGUMENT => [
+                            'Car' => 'something',
+                        ],
+                    ],
+                ]
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    /**
+     * @param MethodCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        foreach ($this->methodCallRemoverArgument as $className => $methodName) {
+            if (! $this->isObjectType($node->var, $className)) {
+                continue;
+            }
+
+            if (! $this->isName($node->name, $methodName)) {
+                continue;
+            }
+
+            // todo resolve variable name
+            return new Variable('this');
```

Comment:
Maybe this would help: https://github.com/rectorphp/rector/blob/8c5fdca38737e45e55f7fca95d743fece567d29e/src/PhpParser/Node/Manipulator/MethodCallManipulator.php#L191

Make the method `public` if needed.

There 2 more services, that are handy for fluent method calls:

- https://github.com/rectorphp/rector/blob/master/rules/magic-disclosure/src/NodeManipulator/ChainMethodCallRootExtractor.php
- https://github.com/rectorphp/rector/blob/master/rules/magic-disclosure/src/NodeAnalyzer/ChainMethodCallNodeAnalyzer.php

Reply:
Great thx, it is what I was looking for!